src/python/bot/fuzzers/syzkaller/engine.py

```diff
@@ -0,0 +1,205 @@
+
+    syzkaller_path = os.path.join(
+        environment.get_value('BUILD_DIR'), 'syzkaller')
+    if not os.path.exists(syzkaller_path):
+      raise SyzkallerError('syzkaller not found in build')
+
+    binary_path = syzkaller_path + '/bin/linux_arm64'
+    for filename in os.listdir(syzkaller_path + '/bin/linux_arm64'):
+      os.chmod(binary_path + '/' + filename, 0o755)
+
+    # Directory to place new units.
+    new_corpus_dir = self._create_temp_corpus_dir('new')
+
+    corpus_directories = [new_corpus_dir]
+    fuzz_timeout = libfuzzer.get_fuzz_timeout(False, total_timeout=max_time)
+
+    return runner.fuzz(
+        corpus_directories,
+        fuzz_timeout=fuzz_timeout,
+        additional_args=options.arguments,
+        artifact_prefix=reproducers_dir,
+        extra_env=options.extra_env)
+
+  def reproduce(self, target_path, input_path, arguments, max_time):
+    """Reproduce a crash given an input.
+
+    Args:
+      target_path: Path to the target.
+      input_path: Path to the reproducer input.
+      arguments: Additional arguments needed for reproduction.
+      max_time: Maximum allowed time for the reproduction.
+
+    Returns:
+      A ReproduceResult.
+    """
+    raise NotImplementedError
+
+  def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
+                      reproducers_dir, max_time):
+    """Optional (but recommended): run corpus minimization.
+
+    Args:
+      target_path: Path to the target.
+      arguments: Additional arguments needed for corpus minimization.
+      input_dirs: Input corpora.
+      output_dir: Output directory to place minimized corpus.
+      reproducers_dir: The directory to put reproducers in when crashes are
+          found.
+      max_time: Maximum allowed time for the minimization.
+
+    Returns:
+      A FuzzResult object.
+    """
+    raise NotImplementedError
+
+  def minimize_testcase(self, target_path, arguments, input_path, output_path,
+                        max_time):
+    """Optional (but recommended): Minimize a testcase.
+
+    Args:
+      target_path: Path to the target.
+      arguments: Additional arguments needed for testcase minimization.
+      input_path: Path to the reproducer input.
+      output_path: Path to the minimized output.
+      max_time: Maximum allowed time for the minimization.
+
+    Returns:
+      A ReproduceResult.
+    """
+    raise NotImplementedError
+
+  def cleanse(self, target_path, arguments, input_path, output_path, max_time):
+    """Optional (but recommended): Cleanse a testcase.
+
+    Args:
+      target_path: Path to the target.
+      arguments: Additional arguments needed for testcase cleanse.
+      input_path: Path to the reproducer input.
+      output_path: Path to the cleansed output.
+      max_time: Maximum allowed time for the cleanse.
+
+    Returns:
+      A ReproduceResult.
+    """
+    raise NotImplementedError
+
+
+class ReproduceResult(object):
```

Comment:
From lines 181 down, they look copy-pasted and unnecesssary.

Reply:
class ReproduceResult is needed as a return value and fails lint if removed. The other two methods (register and get) are removed.

Reply:
Just return the base engine.ReproduceResult. This shouldn't be duplicated here.

Reply:
Done.